controllers/metal3.io/hostfirmwarecomponents_controller.go

```diff
@@ -156,11 +157,18 @@ func (r *HostFirmwareComponentsReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, fmt.Errorf("could not update hostfirmwarecomponents: %w", err)
 	}
 
+	info.log.Info("retrieving firmware components and saving to resource", "Node", bmh.Status.Provisioning.ID)
 	// Check ironic for the components information if possible
 	components, err := prov.GetFirmwareComponents()
-	info.log.Info("retrieving firmware components and saving to resource", "Node", bmh.Status.Provisioning.ID)
 
+	// Improve error checking to only requeue if necessary
 	if err != nil {
+		msg := err.Error()
+		if strings.Contains(msg, "does not support firmware updates") {
```

Comment:
The way to handle this is to create a public error type with `errors.New()` and check for that with `errors.Is()`. See e.g. `provisioner.ErrNeedsRegistration`.